src/ui/training/TrainingCtrl.ts

```diff
@@ -165,18 +171,23 @@ export default class TrainingCtrl implements PromotingInterface {
     this.chessground.stop()
   }
 
-  public newPuzzle = (feedback: boolean) => {
-    if (feedback) this.showLoading()
-    xhr.newPuzzle()
-    .then(cfg => {
+  public newPuzzle = () => {
+    const onSuccess = (cfg: PuzzleData) => {
       this.init(cfg)
-    })
-    .then(this.onXhrSuccess)
-    .catch(this.onXhrError)
+      this.onPuzzleLoad
+    }
+    loadOfflinePuzzle().then(onSuccess, puzzleLoadFailure)
```

Comment:
Don't use `promise.then(success, failure)` form. It's a bad practice in general concerning promises, and in the rest of the app, we always use this form:
```
promise
.then(success)
.catch(failure)
```